doxygen/contributor_help_pages/getting_started.md

```diff
@@ -57,84 +60,136 @@ Within each of those folders you will find any one of the following folders
+  // This works fine
+  //double b2 = dot_self((A_mat * A_vec).eval());
+  // This fails
+  double b2 = dot_self(A_mat * A_vec);
+  std::cout << b2;
+}
 ```
-Actually delays evaluation when making `x` and produces an object of type
+
+Pasting the above into godbolt will show a compilation error! That's because Eigen does now allow coefficient levels access to product functions. Instead that expression needs to be evaluated before it is accessed elementwise. Alternatively, say we have something such as the below where we pass in an expression for elementwise multiplication.
 
 ```cpp
-Eigen::Product<Eigen::Add<Matrix, Eigen::Product<Matrix, Matrix>>, Eigen::Add<Matrix, Matrix>>
+int main() {
+  Eigen::VectorXd A_vec1 = Eigen::VectorXd::Random(5);
+  Eigen::VectorXd A_vec2 = Eigen::VectorXd::Random(5);
+  double b2 = dot_self(A_vec1.array() * A_vec2.array());
+  std::cout << b2;
+}
 ```
 
-Using lazily evaluated expressions allows Eigen to avoid redundant copies, reads, and writes to our data. However, this comes at a cost.
 
-In (2), when we access the coefficients of `x`, if its type is similar to the wacky expression above we can get incorrect results as Eigen does not guarantee any safety of results when performing coefficient level access on a expression type that transforms its inputs. So `to_ref()` looks at its input type, and if the input type is an Eigen expression that it evaluates the expression so that all the computations are performed and the return object is then safe to access.
+This will appear fine, but in our call to `dot_self`, when we call `A.coeff(i)` twice in the for loop we will end up evaluating the expression twice.
 
-But! Suppose our input is something like
+Now Suppose our input is something like
 
 ```cpp
 Eigen::Matrix<var, Dynamic, 1> A = Eigen::VectorXd::Random(20);
 var b = dot_self(A.segment(1, 5));
 ```
 
-Here, we will have an expression `Eigen::Block<Eigen::MatrixXd>` as the input type. But an expression that is only a view into an object is safe to read coefficent-wise. In this case `to_ref()` knows this and, as long as you used `const auto&` as the object type returned from `to_ref(x)`, then `to_ref(x)` will deduce the correct type `Block` instead of casting to an evaluated vector. We use `const auto&` here even when the output will store an object as C++'s [lifetime](https://en.cppreference.com/w/cpp/language/lifetime) rules allow for this.
+Here, we will have an expression `Eigen::Block<Eigen::MatrixXd>` as the input type. But an expression that is only a view into an object is safe to read coefficent-wise. In this case @ref stan::math::to_ref  knows this and, as long as you used `const auto&` as the object type returned from `to_ref(x)`, then `to_ref(x)` will deduce the correct type `Block` instead of casting to an evaluated vector. We use `const auto&` here even when the output will store an object as C++'s [lifetime](https://en.cppreference.com/w/cpp/language/lifetime) rules allow for this.
```

Comment:
> isn't the input type a reference rather than an object?

The input is a reference to an object, though the sentence uses the word object because the statement is true for objects and references to objects

Rest of these I agree with and fixed